packages/inventory-compliance/src/SystemRulesTable.js

```diff
@@ -184,9 +184,13 @@ class SystemRulesTable extends React.Component {
     }
 
     updatePolicyFilterConfig = () => {
-        const policies = this.props.profileRules.map((p) => (
-            p.profile
-        )).filter((p) => !!p);
+        let policies = this.props.profileRules.filter(({ profile }) => !!profile).map(({ profile }) => (
+            {
+                id: profile.policy ? profile.policy.id : profile.id,
+                name: profile.name
+            }
+        ));
+        policies = [ ...new Map(policies.map(policy => [ policy.id, policy ])).values() ];
```

Comment:
Is there a specific reason, why we need to de-duplicate here (in the frontend)?

In case `SystemRulesTable` is passed duplicate profiles it should rather be de-duplicated where the table is used and not pass in duplicate profiles.

Reply:
There are `profileRules` passed consisting of a profile and its rules. What do you have in mind? Passing them as a property?

Reply:
After a discussion, we can remove this line in favor of an assumption that the backend would keep one test result profile per policy.